operator/builtin/input/file/fingerprint_test.go

```diff
@@ -212,4 +214,49 @@ func TestFingerprintStartsWith(t *testing.T) {
 	}
 }
 
+func TestFingerprintStartsWith_FromFile(t *testing.T) {
```

Comment:
I am not familiar with what the fingerprints and how they work so I don't understand what this test does. Perhaps that's fine and I am not supposed to understand. If you want me to understand though please add some comments to explain what it does :-)

Reply:
It's certainly not meant to be a secret. :) 

I expanded the comments on the `Fingerprint` struct and its `StartsWith` method, and also added a high level explanation on the new test.